presto-main/src/main/java/io/prestosql/connector/system/jdbc/TableJdbcTable.java

```diff
@@ -88,15 +89,18 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
         for (String catalog : filter(listCatalogs(session, metadata, accessControl).keySet(), catalogFilter)) {
             QualifiedTablePrefix prefix = tablePrefix(catalog, schemaFilter, tableFilter);
 
-            if (FilterUtil.emptyOrEquals(typeFilter, "TABLE")) {
-                for (SchemaTableName name : listTables(session, metadata, accessControl, prefix)) {
-                    table.addRow(tableRow(catalog, name, "TABLE"));
-                }
+            boolean includeTables = FilterUtil.emptyOrEquals(typeFilter, "TABLE");
+            boolean includeViews = FilterUtil.emptyOrEquals(typeFilter, "VIEW");
+
+            if (!includeTables && !includeViews) {
```

Comment:
The variables are not dependent on the catalog name, so this could go outside the loop.